File: Dart_Source_Files/main_scripts/difficulty.dart
// ignore_for_file: constant_identifier_names

import 'dart:html';

const Map<String, String> ID_MAP = {
  "facile": "1",
  "normale": "2",
  "difficile": "3"
};

const List<String> SCORES = ["100", "150", "200"];
const List<String> OPTIONS = ["2", "3", "4"];

void main() {
  if (window.localStorage.containsKey("diff")) {
    var id = ID_MAP.entries
        .firstWhere((element) => element.value == window.localStorage["diff"]);

    clickedButtonProcessing(querySelector("#${id.key}") as ButtonElement);
  }

  //set event handlers
  querySelector("#facile")!.onClick.listen(clickedButton);
  querySelector("#normale")!.onClick.listen(clickedButton);
  querySelector("#difficile")!.onClick.listen(clickedButton);
}

void clickedButtonProcessing(ButtonElement b) {
  clearButtons();
  b.classes.add("selected");
  String selected = b.id;
  String selectedValue = ID_MAP[selected]!;

  querySelector("#score")!
    ..text = SCORES[int.parse(selectedValue) - 1]
    ..classes.clear();
  querySelector("#options")!
    ..text = OPTIONS[int.parse(selectedValue) - 1]
    ..classes.clear();

  querySelector("#score")!.classes.add(selected);
  querySelector("#options")!.classes.add(selected);

  window.localStorage["diff"] = selectedValue;
}

void clickedButton(Event e) {
  ButtonElement sender = e.target as ButtonElement;
  clickedButtonProcessing(sender);
}

void clearButtons() {
  querySelectorAll("button").classes.clear();
}

File: Dart_Source_Files/main_scripts/difficulty.js
const ID_MAP = new Map([

  ['easy', 1],
  ['medium', 2],
  ['hard', 3],
  [1, 'easy'],
  [2, 'medium'],
  [3, 'hard'],
]);

const SCORES = ["100", "150", "200"];
const OPTIONS = ["2", "3", "4"];

/**
 * 
 * @param {HTMLButtonElement} b 
 */
function clickedButtonProcessing(b) {
  clearButtons();
  b.classList.add('selected');
  let selected = b.id;
  let selectedValue = ID_MAP.get(selected);

  let scoreElement = document.getElementById("score");
  let optionsElement = document.getElementById("options");

  scoreElement.textContent = SCORES[selectedValue - 1];
  optionsElement.textContent = OPTIONS[selectedValue - 1];
  scoreElement.classList = selected;
  optionsElement.classList = selected;

  window.localStorage.setItem('diff', selectedValue);
}

/**
 * 
 * @param {Event} e 
 */
function clickedButton(e) {

  clickedButtonProcessing(e.currentTarget);
}

function clearButtons() {
  document.querySelectorAll(".diff_button").forEach((el) => {

    el.classList = 'diff_button';
  });
}


function main(){

  let id = window.localStorage.getItem('diff');

  if (id) {

    clickedButtonProcessing(new HTMLButtonElement(document.getElementById(`${ID_MAP.get(id)}`)));
  }

  //set event handlers
  [...document.getElementsByClassName("diff_button")].forEach((el) => {

    el.onclick = clickedButton;
  });
}

//start
main();